app/code/Magento/AdminNotification/Controller/Adminhtml/Notification/AjaxMarkAsRead.php

```diff
@@ -21,17 +44,15 @@ public function execute()
         $notificationId = (int)$this->getRequest()->getPost('id');
         $responseData = [];
         try {
-            $this->_objectManager->create(
-                \Magento\AdminNotification\Model\NotificationService::class
-            )->markAsRead(
-                $notificationId
-            );
+            $this->notificationService->markAsRead($notificationId);
             $responseData['success'] = true;
         } catch (\Exception $e) {
             $responseData['success'] = false;
         }
-        $this->getResponse()->representJson(
-            $this->_objectManager->create(\Magento\Framework\Json\Helper\Data::class)->jsonEncode($responseData)
-        );
+
+        /** @var \Magento\Framework\Controller\Result\Json $resultJson */
+        $resultJson = $this->resultFactory->create(ResultFactory::TYPE_JSON);
```

Comment:
`$this->getResponse()->representJson()` vs `$this->resultFactory->create(ResultFactory::TYPE_JSON)` which is the option that should be considered for usage?